rest-server/src/templates/dockerContainerScript.mustache

```diff
@@ -46,20 +50,27 @@ export PAI_KILL_ALL_ON_COMPLETED_TASK_NUM={{{ jobData.killAllOnCompletedTaskNumb
 
 export PAI_CURRENT_CONTAINER_PORT=$(( (RANDOM % 55535) + 10001 ))
 hdfs dfs -touchz $HDFS_LAUNCHER_PREFIX/$PAI_USERNAME/$PAI_JOB_NAME/tmp/$PAI_TASK_ROLE_NO-$PAI_TASK_INDEX-$PAI_CURRENT_CONTAINER_IP-$PAI_CURRENT_CONTAINER_PORT || exit 1
-while [ `hdfs dfs -ls $HDFS_LAUNCHER_PREFIX/$PAI_USERNAME/$PAI_JOB_NAME/tmp/ | wc -l` -lt  $PAI_TASKS_NUM ]; do
+while [ `hdfs dfs -ls $HDFS_LAUNCHER_PREFIX/$PAI_USERNAME/$PAI_JOB_NAME/tmp/ | grep "/$PAI_JOB_NAME/tmp" | wc -l` -lt  $PAI_TASKS_NUM ]; do
+  printf "%s %s\n" "[INFO]" "Waiting for other containers ..."
   sleep 10
 done
 hdfs dfs -ls $HDFS_LAUNCHER_PREFIX/$PAI_USERNAME/$PAI_JOB_NAME/tmp/ \
   | grep "/$PAI_JOB_NAME/tmp" \
   | grep -oE "[^/]+$" \
   | sort -n -k 2 -t"-" \
   > ContainerList
+if [ "$(cat ContainerList | wc -l)" -ne $PAI_TASKS_NUM ]; then
+  printf "%s %s\n" "[ERROR]" "Not all containers are available, exit ..."
```

Comment:
>printf [](start = 2, length = 6)

better print the content of ContainerList and $PAI_TASK_NUM here to help debugging

Reply:
updated